### ISSUE TITLE
Blink cursor only on insert mode

### ISSUE THREAD

**Issue by jjbooth74** [VSCodeVim/Vim#1105]
Please *thumbs-up* 👍 this issue if it personally affects you! You can do this by clicking on the emoji-face on the top right of this post. Issues with more thumbs-up will be prioritized.

-----

### What did you do?

1. Open file, by default in normal mode.
2. Hit `i` to enter insert mode.

What I want:
- In normal mode: a solid block cursor.
- In insert mode: a blinking line cursor.
- (Bonus) in visual mode: blinking block cursor.

### What did you expect to happen?
Cursor switch from solid to blinking + block to line.

### What happened instead?
Cursor switches from block to line, but does not switch from solid to blink.

I've tried various combinations of `editor.cursorStyle`, `editor.cursorBlinking`, and `vim.useSolidBlockCursor` to attempt to get my desired behavior but I can't get different blink behavior across modes.

### Technical details:

Relevant config:
```
    "editor.cursorStyle": "block",
    "editor.cursorBlinking": "solid",
    "vim.useSolidBlockCursor": false,
```

* VSCode Version: 1.7.2
* VsCodeVim Version: 0.4.4
* OS: OS X 10.11.6

**Comment by johnfn** [comment]
With the following in my settings.json, I get a solid block cursor in normal mode and a blinking line cursor in insert mode:

`    "vim.useSolidBlockCursor": true,`

**Comment by johnfn** [comment]
Wait, nevermind. I see what you're saying - the cursor doesn't blink in insert mode with that setting. Hm.

**Comment by jjbooth74** [comment]
Using these settings:
```
    "editor.cursorStyle": "line",
    "editor.cursorBlinking": "blink",
    "vim.useSolidBlockCursor": true,
```
I get this:
![](http://i.giphy.com/26tklr9o4r50F8w5W.gif)


If I use a Solid Line editor setting, I can get rid of the annoying flashing line in normal mode, but then insert mode doesn't flash.
```
    "editor.cursorStyle": "line",
    "editor.cursorBlinking": "solid",
    "vim.useSolidBlockCursor": true,
```
![](http://i.giphy.com/3oz8xtxvsS15YQdbEc.gif)



The block cursor in both previous examples do not match what I expect it to.  I'd expect it to match these settings:
```
    "editor.cursorStyle": "block",
    "editor.cursorBlinking": "solid",
    "vim.useSolidBlockCursor": true,
```
Which looks like:
![](http://i.giphy.com/l0MYOyambWRITrJZK.gif)

The last GIF is what I would expect normal mode to look like (note: the cursor getting lighter is an artifact of the window losing focus while I stop my GIF recording)

**Comment by johnfn** [comment]
So I take it that your problem is that when you turn on useSolidBlockCursor, you can still see the cursor blinking behind the block cursor?

**Comment by johnfn** [comment]
Should be fixed on master and out in 0.4.7!
